### PR TITLE
Add "input shield" to capture pointer input for reinjection

### PR DESCRIPTION
**Pull request by mikaelpessa**

### CODE REVIEW

File: shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc [hunk 1]
```diff
@@ -126,7 +137,7 @@ void FuchsiaExternalViewEmbedder::SubmitFrame(
     GrDirectContext* context,
     std::unique_ptr<flutter::SurfaceFrame> frame) {
   TRACE_EVENT0("flutter", "FuchsiaExternalViewEmbedder::SubmitFrame");
-
+FML_LOG(WARNING) << "EmbeddedView::FuchsiaExternalViewEmbedder::SubmitFrame";
```
Comment (arbreng) [inline shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc:140]: Remove plz

File: shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc [hunk 2]
```diff
@@ -290,6 +301,8 @@ void FuchsiaExternalViewEmbedder::SubmitFrame(
         if (found_rects == scenic_rects_.end()) {
           auto [emplaced_rects, success] = scenic_rects_.emplace(
               std::make_pair(rect_hash, std::vector<scenic::Rectangle>()));
+
```
Comment (arbreng) [inline shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc:304]: Remove plz

File: shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.h [hunk 3]
```diff
@@ -131,6 +132,37 @@ class FuchsiaExternalViewEmbedder final : public flutter::ExternalViewEmbedder {
     scenic::Material material;
   };
 
+  // Helper class for setting up an invisible rectangle to catch all input. Rejected input will then
+  // be re-injected into the scene below us.
```
Comment (arbreng) [inline shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.h:136]: " into the scene below us." -> "into a suitable platform view controlled by this Engine instance."

File: shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.h [hunk 4]
```diff
@@ -131,6 +132,37 @@ class FuchsiaExternalViewEmbedder final : public flutter::ExternalViewEmbedder {
...
+   public:
+    InputInterceptor(scenic::Session* session) : opacity_node_(session), shape_node_(session) {
+      opacity_node_.SetLabel("Flutter::InputInterceptor");
+      opacity_node_.SetOpacity(0.f);
+
+      // Set the shape node to capture all input. Any unwanted input will be reinjected.
+      shape_node_.SetHitTestBehavior(fuchsia::ui::gfx::HitTestBehavior::kDefault);
+      shape_node_.SetSemanticVisibility(false);
+
+  scenic::Material material(session);
```
Comment (arbreng) [inline shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.h:147]: Looks like you need to run clang-format:  buildtools/linux-x64/clang/bin/clang-format

I just set it up in my vscode project with this line in my settings:
"C_Cpp.clang_format_path": "${workspaceFolder}/../buildtools/linux-x64/clang/bin/clang-format",

Comment (arbreng) [reply shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.h:147]: Alternatively you can use the flutter CI scripts that will format all filetypes:

./ci/format.sh  | patch -p0
./ci/check_gn_format.py --gn-binary third_party/gn/gn --root-directory .

The clang-format suggestion is just for if you want format-on-save

**Review by arbreng — APPROVED** [review]
LGTM with some nits and formatting

File: shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc (hunk 1 above)
Comment (mikaelpessa) [reply shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc]: Done.

**Pull request merged**